x/sequencer/keeper/hook_listener.go

```diff
@@ -20,7 +23,13 @@ func (k Keeper) RollappHooks() rollapptypes.RollappHooks {
 	return rollappHook{k: k}
 }
 
-func (hook rollappHook) BeforeUpdateState(ctx sdk.Context, seqAddr string, rollappId string) error {
+// BeforeUpdateState checks various conditions before updating the state.
+// It verifies if the sequencer has been registered, if the rollappId matches the one of the sequencer,
+// if there is a proposer for the given rollappId, and if the sequencer is the active one.
+// If the lastStateUpdateBySequencer flag is true, it also checks if the rollappId is rotating and
+// performs a rotation of the proposer.
+// Returns an error if any of the checks fail, otherwise returns nil.
+func (hook rollappHook) BeforeUpdateState(ctx sdk.Context, seqAddr, rollappId string, lastStateUpdateBySequencer bool) error {
 	// check to see if the sequencer has been registered before
 	sequencer, found := hook.k.GetSequencer(ctx, seqAddr)
```

Comment:
why get the sequencer?
no need
just get the proposer for the rollapp and check the addr matches the msg addr